src/utils/SdkCoin.sol

```diff
@@ -0,0 +1,124 @@
+     * @param tokenAddress The address of the token contract
+     * @return decimals The number of decimals of the token
+     */
+    function _getERC20TokenDecimals(address tokenAddress) internal view returns (uint8) {
+        // Input validation
+        if (tokenAddress == address(0x0)) {
+            revert IISdkCoinErrors.ZeroAddress(tokenAddress);
+        }
+        // If the tokens extends the IERC20 it should be using IERC20Metadata which supports the decimals() call
+        // Why this? -->  https://detectors.auditbase.com/decimals-erc20-standard-solidity
+        try IERC20Metadata(tokenAddress).decimals() returns (uint8 decimals) {
+            return decimals;
+        } catch {
+            // If the decimals function is not implemented, return the default value
+            return DEFAULT_ERC20_DECIMALS;
+        }
+    }
+
+    // Convert ERC20 tokens decimals to Cosmos decimals units
+    // NOTE that the refund to the user of the remainder of the conversion should happen during the transfer.
+    /**
+     * @notice Convert the uint256 ERC20 token amount into cosmos coin amount uint64
+     * @notice This functions allows us to consider the loss of precision
+     * @param tokenAddress The address of the token contract
+     * @param amount The amount to be converted
+     * @return convertedAmount The amount converted to uint64 supported by cosmos coins
+     * @return remainder The remainder of the conversion
+     */
+    function _convertERC20AmountToSdkCoin(
+        address tokenAddress,
+        uint256 amount
+    )
+        internal
+        view
+        returns (uint64, uint256)
+    {
+        // Note if we want to allow different cosmos decimals, we need to handle that here too
+        // Note that tokenAddress input validations are executed in the _getERC20TokenDecimals function
+        uint8 tokenDecimals = _getERC20TokenDecimals(tokenAddress);
+
+        if (tokenDecimals < 6) {
+            revert IISdkCoinErrors.UnsupportedTokenDecimals(tokenDecimals);
+        }
+        // Amount input validation
+        if (amount == 0) {
+            revert IISdkCoinErrors.ZeroAmountUint256(amount);
+        }
+        // Ensure the amount respects the token's decimals
+        // Handle the case where the input amount exceeds the token's precision
+        uint256 temp_convertedAmount;
+        uint256 factor;
+        uint256 remainder;
+        // Case ERC20 decimals are bigger than cosmos decimals
+        if (tokenDecimals > DEFAULT_COSMOS_DECIMALS) {
+            factor = 10 ** (tokenDecimals - DEFAULT_COSMOS_DECIMALS);
+            temp_convertedAmount = amount / factor;
+            remainder = amount % factor;
+        } else if (tokenDecimals == DEFAULT_COSMOS_DECIMALS) {
+            temp_convertedAmount = amount;
+            remainder = 0;
+        } else {
+            // revert as this is unreachable
+            revert IISdkCoinErrors.Unsupported();
```

Comment:
```suggestion
            revert Unsupported();
```
Prefix can be removed if you have the contract inherit the error interface. Please do this to all custom errors in this file